### PR TITLE
Deduplicate Delegated Routing outputs

### PR DESCRIPTION
**Pull request by ischasny**
Records returned from IPNI via Delegated Routing don't have ContextID in them. Becuase of that, some records that are valid from the IPNI point of view might look like duplicates from the Delegated Routing point of view. To make the Delegated Routing output nicer, deduplicate identical records.

Fixes https://github.com/ipni/storetheindex/issues/1760

### CODE REVIEW

File: delegated_translator.go
Comment (willscott) [inline delegated_translator.go:135]: can we have this be a `String()` method on `drProvider`?

Comment (ischasny) [reply delegated_translator.go:135]: I thought about that too initially. The reason I've done it not in the `String` method - is because this is a specific string representation that exist for one specific purpose (deduplication). I.e. it's not a generic string representation of the object (which to me `String` is). But I've been lacking a second opinion. Happy to move it to `drProvider.String` if you feel that it's going to fit it better.

Comment (masih) [reply delegated_translator.go:135]: Should we hash this value to an int via crc32 or something similar. As is the memory footprint seems large for what it's doing.

As an example, see https://github.com/ipni/indexstar/blob/main/find_ndjson.go#L44

Comment (ischasny) [reply delegated_translator.go]: Fair point - fixed

**Comment by ischasny** [comment]
@willscott @masih please approve if there is no more comments

**Review by willscott — APPROVED** [review]

**Review by masih — APPROVED** [review]

**Pull request merged**
